package/MDAnalysis/coordinates/memory.py

```diff
@@ -0,0 +1,242 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- http://www.MDAnalysis.org
+# Copyright (c) 2006-2015 Naveen Michaud-Agrawal, Elizabeth J. Denning, Oliver Beckstein
+# and contributors (see AUTHORS for the full list)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+Reading trajectories from memory --- :mod:`MDAnalysis.coordinates.memory`
+==========================================================================
+
+:Author: Wouter Boomsma
+:Year: 2016
+:Copyright: GNU Public License v2
+:Maintainer: Wouter Boomsma <wb@bio.ku.dk>, wouterboomsma on github
+
+
+.. versionadded:: 0.15.0
+
+The module contains a trajectory reader that operates on an array
+in memory, rather than reading from file. This makes it possible to
+use operate on raw coordinate using existing MDAnalysis tools. In
+addition, it allows the user to make changes to the coordinates in
+a trajectory (e.g. through AtomGroup.set_positions) without having
+to write the entire state to file.
+
+
+Examples
+--------
+
+Constructing a Reader from a numpy array
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A simple example where a new universe is created from the
+array extracted from a DCD timeseries
+
+    from MDAnalysis import Universe
+    from MDAnalysisTests.datafiles import DCD, PDB_small
+    from MDAnalysis.coordinates.array import ArrayReader
+
+    universe = Universe(PDB_small, DCD)
+    coordinates = universe.trajectory.timeseries(universe.atoms)
+
+    universe2 = Universe(PDB_small, coordinates,
+                         format=MemoryReader)
+
+This two step process can also be done in one go:
+
+    universe = Universe(PDB_small, DCD, in_memory=True)
+
+"""
+
+import errno
+import numpy as np
+
+from . import base
+
+
+class MemoryReader(base.ProtoReader):
+    """
+    A trajectory reader interface to a numpy array of the coordinates.
+    For compatibility with the timeseries interface, support is provided for
+    specifying the order of columns through the format option.
+
+    Parameter
+    ---------
+    filename : str
+        filename of the trajectory
+    n_atoms : int
+        number of atoms to write
+    convert_units : bool (optional)
+        convert into MDAnalysis units
+    precision : float (optional)
+        set precision of saved trajectory to this number of decimal places.
+    """
+
+    format = 'memory'
+
+    class MemoryTimestep(base.Timestep):
```

Comment:
@jbarnoud My reasoning was that it doesn't really provide an extension of the interface, and as such, the user can be completely ignorant of its existence. It therefore seemed unnecessary to pollute the namespace with this extra class. However, I don't have any strong opinions on this, and would be happy to move it outside the class if you prefer that solution.